File: Add Code Here/LeetCode_Sols/Easy/104. Maximum Depth of Binary Tree/Recursion - 1/Solution.ex
defmodule TreeNode do
  defstruct val: 0, left: nil, right: nil
end

defmodule Solution do
  def max_depth(nil), do: 0
  def max_depth(%TreeNode{left: left, right: right}) do
    1 + max(max_depth(left), max_depth(right))
  end
end

File: Add Code Here/LeetCode_Sols/Easy/104. Maximum Depth of Binary Tree/Recursion - 1/Solution.py
class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        return self.traverse(root, 0)

    def traverse(self, node: Optional[TreeNode], depth) -> int:
        return depth if not node else max(self.traverse(node.left, depth), self.traverse(node.right, depth)) + 1